developer_tests/namelist/test_get_state_variables.f90

```diff
@@ -0,0 +1,78 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+
+program test_get_state_variables
+
+use        utilities_mod, only : find_namelist_in_file, check_namelist_read 
+use    mpi_utilities_mod, only : initialize_mpi_utilities, finalize_mpi_utilities
+use            types_mod, only : vtablenamelength, MISSING_R8
+use    default_model_mod, only : get_state_variables, state_var_type
+
+use test ! fortran-testanything
+
+implicit none
+
+integer :: iunit, io
+logical :: use_clamping
+type(state_var_type) :: state_vars, state_vars_clamp
+
+! DART state vector contents are specified in the input.nml:&model_nml namelist.
+integer, parameter :: MAX_STATE_VARIABLES = 20
+integer, parameter :: NUM_STATE_TABLE_COLUMNS = 3
+character(len=vtablenamelength) :: state_variables(MAX_STATE_VARIABLES * NUM_STATE_TABLE_COLUMNS ) = ' '
+
+namelist /model_nml_noclamp/  &
+   state_variables
+
+namelist /model_nml_clamp/  &
+   state_variables
+
+call initialize_mpi_utilities('test_get_state_variables')
+
+call plan(10)
+
+! Using namelist WITHOUT clamping values
+
+call find_namelist_in_file('input.nml', 'model_nml_noclamp', iunit)
+read(iunit, nml = model_nml_noclamp, iostat = io)
+call check_namelist_read(iunit, io, 'model_nml_noclamp')
+
+call get_state_variables(state_variables, MAX_STATE_VARIABLES, state_vars)
+
+write(*,*) 'state_vars%nvars: ', state_vars%nvars
+write(*,*) 'state_vars%netcdf_var_names: ', state_vars%netcdf_var_names
+write(*,*) 'state_vars%qtys: ', state_vars%qtys
+write(*,*) 'state_vars%updates: ', state_vars%updates
+
+call ok(state_vars%nvars == 5) !OK
+call ok(state_vars%netcdf_var_names(2) == '') !NOT OK
```

Comment:
Yes, it's supposed to fail. We discussed this a bit, but I can still change it if you'd rather they all pass. Otherwise, I will update the comment to be clearer

Reply:
why not have 
```
call ok(state_vars%netcdf_var_names(2) /= '') 
```
Check for not equal?